addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2270,7 +2270,7 @@ class SnippetsMenu extends Component {
             this._defLoadSnippets = cacheSnippetTemplate[this.options.snippets];
             return this._defLoadSnippets;
         }
-        const context = Object.assign({}, this.options.context);
+        const context = Object.assign({}, this.options.context, {rendering_bundle: true});
```

Comment:
To solve a bug that happens with snippets in edit mode.
First a bit of context: 3rd-party content shouldn't be blocked for website designers (see [this comment from RDE](https://github.com/odoo/odoo/pull/144624#issuecomment-1927567187) for the full context). So everything is made so that the checks are bypassed for that group, and you don't end up as an admin with banners of blocked cookies when outside the editor mode, and you should obviously be able to see the blocks you drop from the editor view.

The bug here is that, once in edit mode, all the snippets are stored in the HTML, and the iframes' `src` are overridden to use `data-src`. So when you drop one of those blocks, the iframe never loads (and it triggers further issues down the line).
I could add `and not request.env.user.has_group('website.group_website_designer')` to the conditions in `_post_processing_att()` and I think that would have a similar effect, but it was first removed (see [this other comment](https://github.com/odoo/odoo/pull/144624#discussion_r1478283784)) because other conditions a few lines above already check some website_designer access. It just happens that the snippets generation wasn't flagged at that point, so it goes through and modifies the `src`, etc. It made sense to flag them as a `rendering_bundle`.

Reply:
> It made sense to flag them as a rendering_bundle.

Indeed, I am actually wondering why this is not a problem already in stable version (for other things that `_post_processing_att` already does). But I guess it's not worth losing time wondering :+1: